Description: Make sure to refresh menu/panel-launchers at least once (when
 the Cinnamon icon theme is the same as the GTK one we don't get any
 icon-theme-change from St)
Author: Clement Lefebvre <clement.lefebvre@linuxmint.com>
Origin: upstream
---
 files/usr/share/cinnamon/applets/menu@cinnamon.org/applet.js        | 6 ++++++
 .../share/cinnamon/applets/panel-launchers@cinnamon.org/applet.js   | 5 +++++
 2 files changed, 11 insertions(+)

diff --git a/files/usr/share/cinnamon/applets/menu@cinnamon.org/applet.js b/files/usr/share/cinnamon/applets/menu@cinnamon.org/applet.js
index 5910184..84a0dda 100644
--- a/files/usr/share/cinnamon/applets/menu@cinnamon.org/applet.js
+++ b/files/usr/share/cinnamon/applets/menu@cinnamon.org/applet.js
@@ -1067,6 +1067,12 @@ MyApplet.prototype = {
             this._pathCompleter.set_dirs_only(false);
             this.lastAcResults = new Array();
             this.settings.bindProperty(Settings.BindingDirection.IN, "search-filesystem", "searchFilesystem", null, null);
+
+            // We shouldn't need to call refreshAll() here... since we get a "icon-theme-changed" signal when CSD starts.
+            // The reason we do is in case the Cinnamon icon theme is the same as the one specificed in GTK itself (in .config)
+            // In that particular case we get no signal at all.
+            this._refreshAll();
+
             St.TextureCache.get_default().connect("icon-theme-changed", Lang.bind(this, this.onIconThemeChanged));
             this._recalc_height();
         }
diff --git a/files/usr/share/cinnamon/applets/panel-launchers@cinnamon.org/applet.js b/files/usr/share/cinnamon/applets/panel-launchers@cinnamon.org/applet.js
index 4abefd5..0b0251d 100644
--- a/files/usr/share/cinnamon/applets/panel-launchers@cinnamon.org/applet.js
+++ b/files/usr/share/cinnamon/applets/panel-launchers@cinnamon.org/applet.js
@@ -305,6 +305,11 @@ MyApplet.prototype = {
 
             this.do_gsettings_import();
 
+            // We shouldn't need to call reload() here... since we get a "icon-theme-changed" signal when CSD starts.
+            // The reason we do is in case the Cinnamon icon theme is the same as the one specificed in GTK itself (in .config)
+            // In that particular case we get no signal at all.
+            this.reload();
+
             St.TextureCache.get_default().connect("icon-theme-changed", Lang.bind(this, this.reload));
         }
         catch (e) {
-- 
1.9.1

